### ISSUE TITLE
Touch foil update

### ISSUE THREAD

**Issue by wibimaster** [zaclittleberry/building-a-touch-mirror#1]
Hi !

Any updates about touch foil and raspberry pi ?

Same as you, some resellers tell me that it's okay with Ubuntu, but cannot say if it was tested with an ARM or Intel architecture :/

Thanks !

**Comment by zaclittleberry** [comment]
Hi @wibimaster , no sorry. I ended up not going the touch foil route because it was very costly for the size monitor I already had. Touch foils seem to come in some common sizes where they cost less. If I recall correctly, up to 7 inches, and then I think 22 inches and up. I had a 20-21 inch monitor so it fell inbetween the ideal sizings. The other consideration was ensuring it to run with a raspberry pi/ARM architecture which several makers responded theirs wouldn't. And then the ever present issue of a touch screen on a mirror will result in a smudgy mirror.

Right now my mirror is not interactive, but eventually I would like to add interaction via a usb webcam. This should be a more versatile and cost effective input device.

**Comment by zaclittleberry** [comment]
I found an email response I sent to someone else with a similar question:

I ended up not going with the touch foil approach because my pre-existing monitor/mirror size (approximately 18.25" x 10.75" in size) ended up being in a 'deadzone' for affordable touch foils. I plan on going for a gesture recognition approach instead. If you are planning out this project without pre-existing requirements and want to use a touch foil, I suggest going with a slightly larger monitor. I think 22" and 24" touch foils are more affordable/readily available.


Here are some companies I found when doing research:
https://www.dhgate.com/price/interactive-touch-foil-price.html
https://www.dhgate.com/product/2-point-22-quot-interactive-touch-screen/395770264.html#se1-7-1b;price|2775733189
http://zytronic.co.uk/ (seems promising if they'd respond)
http://displax.com/
http://cdn2.hubspot.net/hubfs/364008/Touch_Foil_visualplanet_brochure.pdf?t=1463085216389

I also contacted ssidisplay/Profoils/touchfoil.us for $839, which was of course not in budget.

I think I also contacted   http://www.advancedcomponents.co.uk but their touchfoils start at 22", and their configuration only ran on desktop windows or linux (I used a raspberry pi).

I contacted xinttec.com, but their touchfoil required ubuntu and their communication was a little difficult.

A piece of social engineering advice I got from a local hackerspace on getting a better response rate from these companies is to talk as if you are building a prototype, or want to try out a product initially before making more. They are more likely to respond if they think there might be more orders down the line, than if they can only expect one order from you (many of these companies work with other companies that install many of these devices at a time. hobbyists get a low priority).

---Gesture Based----

After the touch film I planned on using a leap motion, however then discovered that the raspberry pi/arm architecture was not supported. Since then I have planned to use a basic usb webcam with open source software for gesture recognition to control the smart mirror, but I haven't gotten back around to it yet.

Hope something in there was useful to you. and best of luck!

**Comment by wibimaster** [comment]
Wow thanks, complete answer :)

It's funny to see how R&D led us in the same directions ^^

I'm using a 24" MVA screen (affordable, 3000:1 contrast, deep is better in darker room).
As a mirror, I'm using a Mirroview 70/20, good light transmission (but not too much), and no metallic coat (otherwise touch foil should fail).

I got a leap motion too, a friend gave me one for free, but never test on a Pi ; it doesn't seem to be designed to be placed in the same alignment as a screen, rather lay flat below or directly on our forehead :/

Gesture based with a camera on a Pi seems to be tricky too, low frame rate when using machine learning process...

I just found https://touchexplorer.en.alibaba.com/ ; they offer custom touch foils, 24 "cost 100 $ with shipping fees (to France). They claims that it's compatible with ARM architecture and RPI.

The only other way is IR frame, it's low cost (around 80$ or less) and works on a Pi, but it leads to a space (1 cm) between mirror and wooden decorative frame on top...

Always hesitate, don't want to throw 100$ on garbage :D

**Comment by wibimaster** [comment]
Finally I bought it :x 

After some discussion with "Touch Explorer Photoelectric Co., Ltd" ("Crystal Chen on Alibaba), they accept to send me the XTM-24S for 50$ + shipping. Hmpf, shpping double the price but... Too much curiosity, I really would like to try it ^^

They confirm Raspberry compatibility, ARM architecture, 10 touch points.

Will tell you how it works :)

Thanks !

**Comment by Rkubinski** [comment]
Hey guys! I am going to most likely go down this route as well (pilkington plus these Alibaba guys for the foil (21.5 inches)), as I saw a similar post to yours on magic mirror. Thanks for investigating so thoroughly :) 

Just a question, why did you pick 70/20 mirror vs their 50/50 mirror  ? Also, how did you order from them, I cant find a form on their site ...
How are things working today ?

**Comment by wibimaster** [comment]
Hi @Rkubinski !

For light transmission and reflection, it depends on what you want to build ; 50/50 is VERY transmitting light and colors, it's better for a mirror in front of a TV for example. It's more a decorative thing when TV is turn off, with good colors transmission when it's turn on.

But in my case, I want to have the mirror in front of a computer screen, and I don't want to turn off the screen every time I don't use it (too much time to turn on). And with a 50/50 mirror, even with a black screen, VA, contrast and luminosity at low level, the backlight is visible in dark room...

Also, I want a really good reflection to use it on a bathroom. 50/50 is a little translucent, decorative but not optimal for mirror default use case ^^

About your second question, I bought it from a german website because I live in France ;)

**Comment by Rkubinski** [comment]
Cool, I am not really looking to have it as a TV so I will go for 70/20.
Did you manage to connect everything and have the touch interface working with the mirroview ?

**Comment by Rkubinski** [comment]
@wibimaster @zaclittleberry do you guys have any instruction manuals on how to get the foil working with a linux ubuntu system or a raspberry system ? I got my foil but cant find any on Ali Baba, I contacted Crystal Chen and should get some from her but was wondering if you have any input.

**Comment by Rkubinski** [comment]
Hey, I got the file working with my laptop Ubuntu system, but when running it with Raspberry, I cant seem to get it working, any ideas ? @wibimaster

**Comment by TheTyrius** [comment]
> About your second question, I bought it from a german website because I live in France ;)

This would be a great option for me as well, can you share the website you went with? Did you get 4mm or 6mm?

Also I would be eager to hear how the combination of this mirrorglas with a touchfoil is working for you (is it accurate and "fun" to use?). Thank you for sharing! :)

**Comment by Rkubinski** [comment]
@TheTyrius I ended up getting 6mm glass. The touch foil works great and is super responsive. However, the touch foil I ordered from Touch Explorer Photoelectric Co., Ltd didnt match in terms of the size. I ordered a 22 inch touch foil and it ended up being an inch thinner or so. It is quite fun to use, however every touch leaves a smudge on the mirror, so I recommend you get a touch pen or some sort of similar device. Another alternative is to use voice commands. If I could re-build my mirror, I would probably still use the touch foil though.

**Comment by TheTyrius** [comment]
@Rkubinski Thanks. I am not very familiar with touch-foils. If it was smaller than expected, is it possible to callibrate it, so the touch location is accurate? (So it can be used even if the foil does not match the display size?).

If I read correctly MirrorView has 25% transparency, which is towards the maximum recommended to not let too show electronics or backlight too much. Does it expose any internals for you or are they adequately hidden by during normal ambient lighting? Did multi-touch work ootb?

**Comment by Rkubinski** [comment]
@TheTyrius Yes it was, I am able to calibrate it so the touch is accurate. Multi touch works, you just have to code for it. You can use Magic Mirror modules that are pre built, but I decided to build my own ElectronJS app from scratch. 

For the transparency -> during the day/ in high lighting settings, it is perfect, and you only see the digits/ letters displayed by the screen. During the night, the backlight of the monitor shines through the mirror. Which is OK for me, I have been thinking of scaling the screen brightness based on ambient lighting, but that would require a camera near the mirror and will be a future project.
